report_print.rb

```diff
@@ -62,6 +62,40 @@ def print_report_summary(report)
   puts
 end
 
+def print_report_motd(report, motd_path)
+  motd = []
+  header = "# #{report.host} #"
+  headline = "#" * header.size
+  motd << headline << header << headline << ''
+
+  motd << "Last puppet run happened at %s in environment %s." % [report.time, report.environment]
+
+  motd << "The result of this puppet run was "
+  case report.status
+  when 'failed'
+    motd.last << color(:red, report.status)
```

Comment:
temp files + atomic writes are a solved problem and only works on the same file system, there are libs for this.

anyway, its fine, if you could remove these `motd.last <<` stuff as above we can do the temp file later